scripts/common/smoketest.go

```diff
@@ -0,0 +1,155 @@
+package common
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+type SmokeTest []*exec.Cmd
+
+type SmokeTestOpts struct {
+	LogDir     string
+	RepoDir    string
+	NumWorkers int
+	Compile    func() error
+	Run        func() error
```

Comment:
Thought: maybe this binary is the orchestrator (bring up the cluster, and eventually start killing, etc.), but not the runner, so it just starts the cluster...

Anyway, that's a bigger refactor and possibly unnecessary, but a thought.